src/test/java/com/backbase/billpay/fiserv/enrolment/EnrolmentControllerTest.java

```diff
@@ -54,5 +72,53 @@ public void postBillPayEnrolment() throws Exception {
         SubscriberEnrollRequest request = retrieveRequest(SubscriberEnrollRequest.class);
         assertHeader("UniqueCustomerNumber", request.getHeader());
     }
+    
+    @Test
+    public void getUserById() throws Exception {
+        
+        // create the mock response
+        SubscriberInfoResponse fiservResponse = SubscriberInfoResponse.builder()
+            .result(ResultType.builder()
+                .success(true)
+                .build())
+            .subscriberInformation(SubscriberInformation.builder()
+                .usAddress(UsAddress.builder()
+                    .address1(ADDRESS_1)
+                    .address2(ADDRESS_2)
+                    .city(CITY)
+                    .state(STATE)
+                    .zip5(ZIP)
+                    .build())
+                .dayPhone(DAY_PHONE)
+                .emailAddress(EMAIL)
+                .name(IndividualName.builder()
+                    .first(FIRST)
+                    .last(LAST)
+                    .prefix(PREFIX)
+                    .build())
+                .taxId(TAX_ID)
+                .build())    
+            .build();
+
+        // set up mock server to return the response
+        setupWebServiceResponse(fiservResponse);
+        
+        // call the endpoint
+        String stringResponse = mockMvc.perform(get(URL + "/" + SUBSCRIBER_ID)
+                        .with(setRemoteAddress())
+                        .param("subscriberID", SUBSCRIBER_ID))
+                        .andExpect(status().isOk())
+                        .andReturn().getResponse().getContentAsString();
+        
+        UserByIdGetResponseBody response =
+                        objectMapper.readValue(stringResponse, UserByIdGetResponseBody.class);
+
+        // validate the subscriber returned matches
+        assertEquals(FIRST, response.getSubscriber().getFirstName());
```

Comment:
Can all of the fields be asserted to ensure the mapping is tested?

Reply:
They sure can. Done